python/tchannel/zipkin/tracers.py

```diff
@@ -126,27 +133,25 @@ class TChannelZipkinTracer(object):
     buffering of any sort.
     """
 
-    def __init__(self, tchannel):
+    def __init__(self, tchannel, routers):
         """
         :param tchannel:
             A tchannel instance to send the trace info to zipkin server
+        :param routers:
+            A list contains hyperbahn instances' ip addresses
         """
         self._tchannel = tchannel
-        self.client = tchannel.request(
-            "tcollector"
-        )
+        self.client = TCollectorClient(self._tchannel)
+
+        if routers:
+            hyperbahn.advertise(self._tchannel, 'tcollector', routers)
```

Comment:
Why are you advertising as 'tcollector'? That's the service you're calling. You only advertise yourself as the service you provide.